sktime/forecasting/croston.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+import pandas as pd
+from sktime.forecasting.base._sktime import _SktimeForecaster
+from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
+
+
+DEFAULT_SMOOTHING = 0.1
+DEFAULT_ALPHA = 0.05
+
+
+class Croston(_OptionalForecastingHorizonMixin, _SktimeForecaster):
+    """Croston's Forecasting Method.
+
+    This was designed for forecasting intermittent demand.
+
+    Parameters
+    -----------
+    smoothing : float, default = 0.1
+        Smoothing parameter
+
+    Examples
+    --------
+    >>> from sktime.forecasting.croston import Croston
+    >>> from sktime.datasets import load_PBS_dataset
+    >>> y = load_PBS_dataset()
+    >>> forecaster = Croston(smoothing=0.1)
+    >>> forecaster.fit(y)
+    Croston(...)
+    >>> y_pred = forecaster.predict(fh=[1,2,3])
+
+    References
+    ----------
+    [1]  J. D. Croston. Forecasting and stock control for intermittent demands.
+        Operational Research Quarterly (1970-1977), 23(3):pp. 289–303, 1972.
+    [2]  Forecasting: Principles and Practice,
+        Otext book by Rob J Hyndman and George Athanasopoulos
+    """
+
+    def __init__(self, smoothing=DEFAULT_SMOOTHING):
+        # hyperparameter
+        self.smoothing = smoothing
+        self.f = None
+        super(Croston, self).__init__()
+
+    def fit(self, y, X=None, fh=None):
```

Comment:
Can I ask you to refactor this one more time? We did some refactoring in the mean time, please see #912 and especially the extension template in `sktime/extension_templates/forecasting.py`.